### PR TITLE
Add e2e test for optional JWT with authZ

### PR DESCRIPTION
**Pull request by diemtvu**
E2E tests for combination of optional JWT (`originIsOptional=true`) and authorization rules.

### CODE REVIEW

File: tests/integration/security/optional_jwt_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+						Options: echo.CallOptions{
+							Target:   b,
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+						},
+					},
+					ExpectAuthenticated: false,
+				},
+				{
+					Name: "accessing-c-include-path-without-token",
```
Comment (yangminzhu) [inline tests/integration/security/optional_jwt_test.go]: Could you add a case for `accessing-c-include-path-with-token`?

File: tests/integration/security/optional_jwt_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	"istio.io/istio/tests/integration/security/util/authn"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+const (
+	authHeaderKey = "Authorization"
+)
+
+// Test authentication policy with optional JWT, and enforce requirement per host, path etc via authZ.
+func TestOptionalAuthnJwt(t *testing.T) {
```
Comment (yangminzhu) [inline tests/integration/security/optional_jwt_test.go]: There is already a TestV1beta1_JWT in `tests/integration/security/rbac_v1beta1_test.go` that covers the `originIsOptional: true`, could you merge this to the other one?

File: tests/integration/security/optional_jwt_test.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+							Target:   c,
+							Path:     "/healthz",
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+						},
+					},
+					ExpectAuthenticated: true,
+				},
+				{
+					Name: "accessing-d-exclude-path-without-token",
```
Comment (yangminzhu) [inline tests/integration/security/optional_jwt_test.go]: What does `exclude` mean here? I don't see any authorization policy for d.

File: tests/integration/security/optional_jwt_test.go (hunk 2 above)
Comment (diemtvu) [reply tests/integration/security/optional_jwt_test.go]: Didn't realize we already have it. I've changed that test to use namespace-level policy and add an instance (`c`) that doesn't have authz to verify the JWT requirements won't be enforced on `c`.

**Pull request merged**